server/datastore/mysql/migrations/tables/20240313143039_TrackUserReferencesForCommandsAndProfiles.go

```diff
@@ -0,0 +1,84 @@
+package tables
+
+import (
+	"database/sql"
+	"fmt"
+)
+
+func init() {
+	MigrationClient.AddMigration(Up_20240313143039, Down_20240313143039)
+}
+
+func Up_20240313143039(tx *sql.Tx) error {
+	// create a new table to store user information that's persisted after
+	// users are deleted.
+	_, err := tx.Exec(`
+	  CREATE TABLE IF NOT EXISTS user_persistent_info (
+	    -- id is an unique identifier for the row, independent from whatever is stored in 'users'
+	    id int(10) unsigned NOT NULL AUTO_INCREMENT,
+
+	    -- user_id is a nullable FK reference to the users table
+	    user_id int(10) unsigned DEFAULT NULL,
+
+	    -- user_name mirrors the users.name value
+	    user_name varchar(255) COLLATE utf8mb4_unicode_ci NOT NULL DEFAULT '',
+
+	    -- timestamps
+	    created_at timestamp NOT NULL DEFAULT CURRENT_TIMESTAMP,
+	    updated_at timestamp NOT NULL DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP,
+
+	    PRIMARY KEY (id),
+	    FOREIGN KEY (user_id) REFERENCES users (id) ON DELETE SET NULL
+	  )
+      `)
+	if err != nil {
+		return fmt.Errorf("failed to add user_persistent_info table: %w", err)
+	}
+
+	// migrate existing data.
+	_, err = tx.Exec(`
+	  INSERT INTO user_persistent_info (user_id, user_name)
+	  SELECT id, name
+	  FROM users
+	`)
```

Comment:
should we drop `users.name` and force people to `JOIN` using this table?

Reply:
Good questions... Would it be very hard to avoid both? I.e. keep the mirrored info (as we also do for `activities` I think?), and avoid the trigger by making sure the updates of `users` also update that table in a transaction?

Reply:
sounds good to me! ended up going with that.